web/src/hooks/nightTimes.ts

```diff
@@ -0,0 +1,72 @@
+import useGetZone from 'api/getZone';
+import {
+  addDays,
+  closestIndexTo,
+  compareAsc,
+  isAfter,
+  isBefore,
+  max,
+  min,
+  subDays,
+} from 'date-fns';
+import { useGetGeometries } from 'features/map/map-utils/getMapGrid';
+import { getSunrise, getSunset } from 'sunrise-sunset-js';
+import { getZoneFromPath } from 'utils/helpers';
+
+export function useNightTimes() {
+  const { data } = useGetZone();
+  const geometries = useGetGeometries();
+  const zoneId = getZoneFromPath();
+
+  if (!zoneId || !data) {
+    return undefined;
+  }
+
+  // get latitude and longitude of zone
+  const feature = geometries.features.find(
+    (feature) => feature.properties.zoneId === zoneId
+  );
+  const coord = feature?.properties.center;
+  if (!coord) {
+    console.warn('no center coordinates available');
+    return undefined;
+  }
```

Comment:
In what scenario can this happen? All zones should have centers calculated and they should always be available when the feature is available.

Reply:
It's only due to using the `.find()` lookup which can return `undefined` if element is not found. Happy to redo that part if you have a better solution in mind :)

Reply:
Hmm, yeah that is a problem. All the solutions that I can think of involve bigger changes.

Either we shift to using the bounding box from the zone config (which we can access directly) and use that to get the center instead.
Or we change how the config files in general are computed in a future version where we can use the center calculated by topojson but insert it into the config instead (at build).

But that seems like a lot of work for just this... (could help us optimize the loading though by shrinking the world file and moving it over to the config instead))